### PR TITLE
Ft testing

### PR DESCRIPTION
**Pull request by bryan936M**
#### What does this PR do?
Implements the testing process on the REST API
#### Description of Task to be completed
- Successfully tested the post API
- Successfully tested the message API
- Made adjustments to route & controller files
#### How should this be manually tested?
- Cloning & run project
#### What are the relevant pivotal tracker stories?
- visit: https://trello.com/invite/b/nLBA9kER/ATTI7620fcbf2169c3045a61a5c97751ff38B437BE1C/andela-portfolio-project
- View 'final steps on backend' card in the doing or done list
#### Screenshots (if appropriate)

### CODE REVIEW

File: controllers/messageController.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const sendMessage = async( req, res) => {
 }
 const updateMessage = async( req, res) => {
     try {
-        if( !req?.body?.id) return res.status(400).json({"message":"ID not present."});
+        if( !req?.body?.id || !mongoose.Types.ObjectId.isValid(req.body.id)) return res.status(400).json({"message":"ID not present."});
```
Comment (chanceDira) [inline controllers/messageController.js:32]: Can you use Jsend format in your response.

File: server.test.js [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+            //check status for a 404 & a message
+            expect(response.status).toBe(404);
+            expect(response.header['content-type']).toMatch(/application\/json/);
+            expect(response.body).toEqual({ message: "There's no data." })
+
+            Post.find.mockRestore();
+        });
+
+        // it('should return a 500 error when there is a server error', async () => {
+        //     const postController = require('./controllers/postController');
```
Comment (chanceDira) [inline server.test.js:37]: If you don't need these codes you remove it.

File: server.test.js [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+    describe('DELETE messages', () => {
+        it('should delete a MESSAGE using ID received from url params', async() => {
+            // Create a sample MESSAGE
+            const messageData = {
+                name: 'Dummy Name',
+                email: 'test@email.com',
+                msg: 'This is a new message',
+            };
+            const message = await Message.create(messageData);
```
Comment (chanceDira) [inline server.test.js:256]: This is good 👍

**Review by chanceDira — CHANGES_REQUESTED** [review]
Good work @M936-B,  just few issues

File: controllers/messageController.js (hunk 1 above)
Comment (bryan936M) [reply controllers/messageController.js:32]: ok, I will.

**Pull request merged**